src/AppInstallerCLICore/Workflows/RepairFlow.cpp

```diff
@@ -459,9 +562,26 @@ namespace AppInstaller::CLI::Workflow
         context <<
             GetManifestWithVersionFromPackage(
                 requestedVersion,
-                context.Args.GetArg(Execution::Args::Type::Channel), false) <<
-            SelectInstaller <<
-            EnsureApplicableInstaller;
+                context.Args.GetArg(Execution::Args::Type::Channel), false);
+    }
+
+    void SelectApplicableInstallerWhenEssential(Execution::Context& context)
+    {
+        // For MSI installers, the platform provides built-in support for repair via msiexec, hence no need to select an installer.
+        // Similarly, for MSIX packages that are not from the Microsoft Store, selecting an installer is not required.
+        if (IsInstallerMappingRequired(context))
+        {
+            // Non Manifest based repair flow
+            if (!context.Args.Contains(Args::Type::Manifest))

```

Comment:
nit: move the check inside SelectApplicablePackageVersion and return if it's manifest flow